transport/tchannel/options.go

```diff
@@ -175,3 +176,10 @@ func ConnBackoff(s backoffapi.Strategy) TransportOption {
 		options.connBackoffStrategy = s
 	}
 }
+
+// ForwardingHeader specifies whether to forward headers without canonicalizing it
+func ForwardingHeader() TransportOption {
```

Comment:
How about we call this `PreserveHeaderNames`? @kriskowal?

The documentation will have to mention that YARPC normalizes header names to lower case by default.